Mathlib/NumberTheory/LocalField/Basic.lean

```diff
@@ -0,0 +1,176 @@
+/-
+Copyright (c) 2025 Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang
+-/
+import Mathlib.Topology.Algebra.Valued.LocallyCompact
+import Mathlib.Topology.Algebra.Valued.ValuedField
+import Mathlib.Topology.Algebra.Valued.ValuativeRel
+import Mathlib.RingTheory.Valuation.RankOne
+
+/-!
+
+# Definition of (Non-archimedean) local fields
+
+Given a topological field `K` equipped with an equivalence class of valuations (a `ValuativeRel`),
+we say that it is a non-archimedean local field if the topology comes from the given valuation,
+and it is locally compact and non-discrete.
+
+-/
+
+/--
+Given a topological field `K` equipped with an equivalence class of valuations (a `ValuativeRel`),
+we say that it is a non-archimedean local field if the topology comes from the given valuation,
+and it is locally compact and non-discrete.
+
+This implies the following typeclasses via `inferInstance`
+- `ValuativeTopology K`
+- `LocallyCompactSpace K`
+- `IsTopologicalDivisionRing K`
+- `ValuativeRel.IsNontrivial K`
+- `ValuativeRel.IsRankLeOne K`
+- `ValuativeRel.IsDiscrete K`
+- `IsDiscreteValuationRing 𝒪[K]`
+- `Finite 𝓀[K]`
+
+Assuming we have a compatible `UniformSpace K` instance
+(e.g. via `IsTopologicalAddGroup.toUniformSpace` and `isUniformAddGroup_of_addCommGroup`) then
+- `CompleteSpace K`
+- `CompleteSpace 𝒪[K]`
+-/
+class IsNonarchLocalField (K : Type*) [Field K] [ValuativeRel K] [TopologicalSpace K] : Prop extends
+  IsTopologicalAddGroup K,
+  ValuativeTopology K,
+  LocallyCompactSpace K,
+  ValuativeRel.IsNontrivial K
```

Comment:
As discussed on Zulip, a constructor that takes
```
lemma IsNonarchLocalField.ofRel {K : Type*} [Field K] [ValuativeRel K] [TopologicalSpace K] [IsTopologicalAddGroup K]
    [ValuativeTopology K] [ValuativeRel.IsNontrivial K]
    [ValuativeRel.IsDiscrete K] [ValuativeRel.IsRankLeOne K] :
    IsNonarchLocalField K :=
```
would be welcome!

Reply:
where are you getting completeness from?